services/src/api/model/responses.rs

```diff
@@ -0,0 +1,169 @@
+use crate::handlers::ErrorResponse;
+use utoipa::ToResponse;
+
+#[derive(ToResponse)]
+#[response(description = "Media type of application/json is expected", example = json!({
+    "error": "UnsupportedMediaType",
+    "message": "Unsupported content type header."
+}))]
+pub struct UnsupportedMediaTypeForJsonResponse(ErrorResponse);
+
+#[derive(ToResponse)]
+#[response(description = "Payload too large", examples(
+    ("Unknown payload size" = (value = json!({
+        "error": "Overflow",
+        "message": "JSON payload has exceeded limit (2097152 bytes)."
+    }))),
+    ("Known payload size" = (value = json!({
+        "error": "Overflow",
+        "message": "JSON payload (XXX bytes) is larger than allowed (limit: 2097152 bytes)."
+    })))
+))]
+pub struct PayloadTooLargeResponse(ErrorResponse);
+
+#[derive(ToResponse)]
+#[response(description = "Id of generated resource", example = json!({
+    "id": "36574dc3-560a-4b09-9d22-d5945f2b8093"
+}))]
+pub struct IdResponse {
+    pub id: String,
+}
+
+#[derive(ToResponse)]
+#[response(description = "Authorization failed", examples(
+    ("Authorization Header is missing" = (value = json!({
+        "error": "MissingAuthorizationHeader",
+        "message": "Header with authorization token not provided."
+    }))),
+    ("Authorization Scheme other than Bearer is used" = (value = json!({
+        "error": "InvalidAuthorizationScheme",
+        "message": "Authentication scheme must be Bearer."
+    }))),
+    ("Provided auth token has an invalid format" = (value = json!({
+        "error": "InvalidUuid",
+        "message": "Identifier does not have the right format."
+    }))),
+    ("Auth token does not correspond to an admin" = (value = json!({
+        "error": "InvalidAdminToken",
+        "message": "Invalid admin token"
+    }))),
+    ("Session id is invalid" = (value = json!({
+        "error": "InvalidSession",
+        "message": "The session id is invalid."
+    }))),
+))]
+pub struct UnauthorizedAdminResponse(ErrorResponse);
+
+#[derive(ToResponse)]
+#[response(description = "Authorization failed", examples(
+    ("Authorization Header is missing" = (value = json!({
```

Comment:
each example should correspond to exactly one Error-Enum-Variant, right? Is it possible to make sure every enum variant has an example? Or can we maybe add the example to the error enum variant instead?

Reply:
Yes, each example should correspond to exactly one possible error variant.

To check if for every possible error has an example a program would need to know which errors can be thrown. This is not possible because errors can contain other errors using the `source` field. For example for the "Authorization" error you do not know which possible inner errors there are because this depends on the appplication logic and cannot be deduced from the error enum.

You cannot derive `ToResponse` for an enum and then add the example to the variant. Utoipa is not able to parse the schema correctly and generates "something".

It is not possible to assert that